src/OrleansProviders/Streams/Common/PooledCache/FixedSizeObjectPool.cs

```diff
@@ -33,8 +33,11 @@ public class FixedSizeObjectPool<T> : ObjectPool<T>
         /// </summary>
         /// <param name="poolSize"></param>
         /// <param name="factoryFunc"></param>
-        public FixedSizeObjectPool(int poolSize, Func<T> factoryFunc)
-            : base(factoryFunc, poolSize)
+        /// <param name="blockPoolId">Unique Id for current block pool</param>
+        /// <param name="monitor"></param>
+        /// <param name="monitorWriteInterval"></param>
+        public FixedSizeObjectPool(Func<T> factoryFunc, string blockPoolId, int poolSize, IObjectPoolMonitor monitor = null, TimeSpan? monitorWriteInterval = null)
```

Comment:
will change block to object, when it is related to object pool.

I  used block mainly because users are more familiar with that term.